lean4/src/putnam_2017_b2.lean

```diff
@@ -11,7 +11,7 @@ theorem putnam_2017_b2
 (S : ℤ → ℕ → ℤ)
 (p : ℤ → ℕ → Prop)
 (q : ℤ → Prop)
-(hS : S = fun a k ↦ ∑ i : Fin k, a + i)
+(hS : S = fun (a : ℤ) k ↦ ∑ i in Finset.range k, (a + i))
```

Comment:
You shouldn't need the type ascription here

Reply:
When I remove it, I get the error:
> type mismatch
  fun a k => ∑ i ∈ Finset.range k, (a + i)
has type
  ℕ → ℕ → ℕ : Type
but is expected to have type
  ℤ → ℕ → ℤ : Type

Reply:
Huh, strange; that wasn't what I expected!